dd-java-agent/appsec/src/main/java/com/datadog/appsec/config/AppSecConfigServiceImpl.java

```diff
@@ -384,13 +508,7 @@ private void applyRemoteConfigListener() {
     setAppSecActivation(features.asm);
     setUserIdCollectionMode(features.autoUserInstrum);
 
-    if (!AppSecSystem.isActive() || !currentAppSecConfig.dirtyStatus.isAnyDirty()) {
-      return;
-    }
-
-    distributeSubConfigurations(
-        Collections.singletonMap("waf", currentAppSecConfig), reconfiguration);
-    currentAppSecConfig.dirtyStatus.clearDirty();
+    distributeSubConfigurations("waf", reconfiguration);
```

Comment:
I think we should keep the previous guard here:

```suggestion
    if (!AppSecSystem.isActive()) {	    
      return;
    }
    
    distributeSubConfigurations("waf", reconfiguration);
```

Otherwise, we'll be triggering libddwaf-java initialization for services where AppSec is not enabled.